### ISSUE TITLE
Composer files - clean up needed

### ISSUE THREAD

**Issue by tmccormi** [LibreHealthIO/LibreEHR#725]
Are these files needed?  we are not "installing" any outside packages using composer builds in the current model to my knowledge. 

./modules/html2pdf/composer.json
./modules/html2pdf/vendor/composer
./modules/html2pdf/vendor/tecnickcom/tcpdf/composer.json
./modules/html2pdf/vendor/setasign/fpdi/composer.json
./modules/html2pdf/composer.lock
./composer.json
./interface/tags_filters/composer.json
./interface/tags_filters/vendor/symfony/process/composer.json
./interface/tags_filters/vendor/composer
./interface/tags_filters/vendor/components/jquery/composer.json
./interface/tags_filters/vendor/kriswallsmith/assetic/composer.json
./interface/tags_filters/vendor/robloach/component-installer/composer.json
./interface/tags_filters/composer.lock
./library/vendor/composer
./library/adodb/composer.json
./assets/js/summernote-0-8-2/composer.json
./composer.lock

**Comment by aethelwulffe** [comment]
Nope.  I have had those files in a cut-list on several PR's in the 
past.  They seem to be everywhere.  Unfortunately, I recall there are 
some files named "composer" that were not necessarily composer.

The habit has been to install a whole complete package from whatever 
junked up set of source files are available, rather than cleaning it up 
and using only the required components.  Too much work I guess.

**Comment by kchapple** [comment]
@tmccormi @aethelwulffe I would keep them, as they do not take up too much space, and do provide information about the versions of these components and their dependencies. And if we ever wanted to upgrade anything using composer we could.

**Comment by aethelwulffe** [comment]
Just my point of view:

Anything that is kept must be maintained, even if it is never used.

Ref: https://github.com/sensiolabs/security-checker

   I am not sure why one would include what are essentially makefiles in 
a deployed product.  The larger issue is that composer, being limited in 
how it manages file locations, should in fact be running on the 
development machine at a higher directory level than the webroot...not 
with the application leaving stuff essentially all over the place, and 
then there are little composer files everywhere in mixed directories.

I find that it supports monolithic architecture fine.  It does not 
specifically support any sort of package module scheme.  There are other 
limits that have demonstrated, to me, any net benefit is restricted to 
enterprise level operations that have single points of authority 
controlling the content.

Upgrading, using composer, seems to be very messy.  On your machine, 
space is not an issue.  For an unfunded distribution machine, it starts 
getting important.

Personally, I think we need to be picky and minimalist when choosing 3rd 
party assets.

I think that if an individual developer wants to utilize composer (which 
is not, and never will be deployed on anything but a few enterprise 
vendor machines, and only Linux: because on windows it I has its own 
prerequisite and permissions dependency hell) then they could and should 
utilize it to paste in the files they want, where they should actually 
go, then leave the actual application code base clean.  The composer 
configuration and dev toolset could, and should go into separate repo.

Just my opinion...

**Comment by teryhill** [comment]
those files could be moved to a place outside the code base as an archive for possible use later if they are needed.

**Comment by kchapple** [comment]
Since we're not using composer as a project at this time, most of these could be deleted. Except the root level one because it is actually used by the plugin system. It was my hope before the Tags/Filter thing was permanently checked into the repo, that it would be installed via composer. I would heavily support a move towards a single composer file and install dependencies via composer.

>    I am not sure why one would include what are essentially makefiles in a deployed product.  

This is the source code repository, not the deployed product. In my 15 years as a professional developer, I've never seen makefiles outside of the source tree. 

> 
I find that it supports monolithic architecture fine.  It does not specifically support any sort of package module scheme.

This is exactly what it does though not in the way we're using it. Perhaps we should move all of these composer packages to a single directory (like vendor) so they can be managed?

> Upgrading, using composer, seems to be very messy.

I don't understand this at all. When used correctly, "composer update" updates all of your modules and dependencies. How is that messy?

> Personally, I think we need to be picky and minimalist when choosing 3rd party assets.

I agree.

> never will be deployed on anything but a few enterprise vendor machines, and only Linux

Composer is a PHP script and runs on any platform, so not sure what you're getting at here.

**Comment by kchapple** [comment]
If this is a permanent feature, I'd suggest moving it out of the Tags plugin and into it's own library file so it won't break anything if the Tags/Filter plugin is removed.

**Comment by Jdew192837** [comment]
Is this something I could do @tmccormi @aethelwulffe @teryhill ? Or does this require further discussion?

**Comment by tmccormi** [comment]
@Jdew192837  Moving it out of the Tags plugin and into it's own library file so it won't break anything if the Tags/Filter plugin is removed.
